packages/core/core/src/services/document-service/utils/repair-orphan-unidirectional-relations.ts

```diff
@@ -0,0 +1,269 @@
+    const targetModel = db.metadata.get(relation.target);
+    if (!targetModel) {
+      db.logger.debug(`Target model ${relation.target} not found, skipping ${joinTableName}`);
+      return 0;
+    }
+
+    // Check if target supports draft/publish using schema-based approach (like prevention fix)
+    const targetContentType = strapi.contentTypes[relation.target as keyof typeof strapi.contentTypes];
+    const targetSupportsDraftPublish = targetContentType?.options?.draftAndPublish || false;
+
+    if (!targetSupportsDraftPublish) {
+      return 0;
+    }
+
+    // Find entries with publication state mismatches
+    const ghostEntries = await findPublicationStateMismatches(
+      db,
+      joinTableName,
+      relation,
+      targetModel,
+      sourceModel
+    );
+
+    if (ghostEntries.length === 0) {
+      return 0;
+    }
+
+    // Remove ghost entries
+    await db.connection(joinTableName).whereIn('id', ghostEntries).del();
+    db.logger.debug(
+      `Removed ${ghostEntries.length} ghost relations with publication state mismatches from ${joinTableName}`
+    );
+
+    return ghostEntries.length;
+  } catch (error) {
+    const errorMessage = error instanceof Error ? error.message : String(error);
+    db.logger.error(`Failed to clean join table "${joinTableName}": ${errorMessage}`);
+    return 0;
+  }
+};
+
+/**
+ * Finds join table entries with publication state mismatches
+ * Uses existing component parent detection from document service
+ */
+const findPublicationStateMismatches = async (
+  db: Database,
+  joinTableName: string,
+  relation: any,
+  targetModel: any,
+  sourceModel: any
+): Promise<number[]> => {
+  try {
+    // Get join column names using proper functions (addressing PR feedback)
+    const sourceColumn = relation.joinTable.joinColumn.name;
+    const targetColumn = relation.joinTable.inverseJoinColumn.name;
+
+    // Get all join entries with their target entities
+    const query = db
+      .connection(joinTableName)
+      .select(
+        `${joinTableName}.id as join_id`,
+        `${joinTableName}.${sourceColumn} as source_id`,
+        `${joinTableName}.${targetColumn} as target_id`,
+        `${targetModel.tableName}.published_at as target_published_at`
+      )
+      .leftJoin(
+        targetModel.tableName,
+        `${joinTableName}.${targetColumn}`,
+        `${targetModel.tableName}.id`
+      );
+
+    const joinEntries = await query;
+
+    // Group by source_id to find duplicates pointing to draft/published versions of same entity
+    const entriesBySource: { [key: string]: any[] } = {};
+    for (const entry of joinEntries) {
+      const sourceId = entry.source_id;
+      if (!entriesBySource[sourceId]) {
+        entriesBySource[sourceId] = [];
+      }
+      entriesBySource[sourceId].push(entry);
+    }
+
+    const ghostEntries: number[] = [];
+
+    // Check if this is a component join table
+    const isComponentJoinTable = joinTableName.match(/^components_.*_.*_lnk$/);
```

Comment:
This seems like it will work, but let's explicitly reference where the strings `components_` and `_lnk` come from, or even better import them if that's possible without refactoring other files or potentially introducing circular references.

Reply:
I can set `collectionName: 'blablabla'` in my component object and it will _not_ be named `components_` in the db.